Marlin/Marlin_main.cpp

```diff
@@ -2962,6 +2990,23 @@ inline void gcode_M17() {
   enable_all_steppers();
 }
 
+#ifdef RESUME_FEATURE
+  inline void gcode_M19() {
+    if(code_seen('Z'))
+    {
+      gcode_get_destination(); // For Z
+      prepare_move();
```

Comment:
Yes! It seems that without a call to `st_synchronize`, the `M114` command would be processed immediately, before the move just added to the planner even starts.